experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1373 @@
+      exp = ev[var_pos]
+      shifted_var_pos = old_to_new_pos[var_pos]
+
+      new_exp_vec = copy(ev)
+      new_exp_vec[var_pos] -= 1
+      new_exp_vec[shifted_var_pos] += 1
+
+      push_term!(C, coeff_t * exp, new_exp_vec)
+    end
+  end
+  return dpr(finish(C))
+end
+
+@doc raw"""
+    diff_action(p::ActionPolyRingElem, d::Vector{Int}) -> ActionPolyRingElem
+
+Successively apply the `i`-th diff-action `d[i]`-times to the polynomial `p`, where $i = 1, \ldots, length(d)$. 
+"""
+function diff_action(dpre::DifferencePolyRingElem{T}, d::Vector{Int}) where {T}
+  dpr = parent(dpre)
+  @req length(d) == ndiffs(dpr) && all(j -> j >= 0, d) "Invalid vector of multiplicities"
+  if is_zero(d)
+    return dpre
+  end
+  dpre_vars_idxs = map(var -> __vtj(dpr)[var], vars(dpre))
+  add_new_vars_idx = filter(new_idx -> !haskey(__jtu_idx(dpr), new_idx), map(idx -> (idx[1], idx[2] + d), dpre_vars_idxs))
+  if !is_empty(add_new_vars_idx)
+    __add_new_jetvar!(dpr, add_new_vars_idx)
+  end
+
+  jtu = __jtu_idx(dpr)
+  old_to_new_pos = Dict{Int, Int}() #Dictionary that links the positions of old with new variables
+  for i in 1:length(dpre_vars_idxs)
+    (i, idx) = dpre_vars_idxs[i]
+    new_idx = jtu[(i, idx + d)]
+    old_to_new_pos[jtu[(i, idx)]] = new_idx
+  end 
+  
+  upr = __upr(dpr)
+  upre = data(dpre)
+  C = MPolyBuildCtx(upr)
+
+  for term in terms(upre)
+    coeff_t = coeff(term, 1)
+    vars_t = vars(term)
+    ev = append!(exponent_vector(term, 1), fill(0, ngens(upr) - length(exponent_vector(term, 1))))
+    new_exp_vec = copy(ev)
+
+    for var in vars_t
+      var_pos = findfirst(==(var), gens(upr))
+      shifted_var_pos = old_to_new_pos[var_pos]
+      
+      new_exp_vec[shifted_var_pos] = ev[var_pos]
+      new_exp_vec[var_pos] = 0 
+    end
+    
+    push_term!(C, coeff_t , new_exp_vec)
+  end
+  return dpr(finish(C))
+end
+
+function diff_action(dpre::DifferentialPolyRingElem{T}, d::Vector{Int}) where {T}
+  len = length(d)
+  @req len == ndiffs(parent(dpre)) && all(j -> j >= 0, d) "Invalid vector of diff multiplicities"
+  res = dpre
+  for i in 1:len
+    for j in 1:d[i]
+      res = diff_action(res, i)
+    end
+  end
+  return res
+end
+
+@doc"""
+    initial(p::ActionPolyRingElem)
+
+Return the initial of the polynomial `p`, i.e. the leading coefficient of `p` regarded as a univariate polynomial in its leader.
+"""
+function initial(apre::ActionPolyRingElem)
+  if is_constant(apre)
+    return apre
+  end
+  return divexact(leading_term(apre), leader(apre)^degree(apre, leader(apre)); check = true)
+end
+
+@doc raw"""
+    leader(p::ActionPolyRingElem)
+
+Return the leader of the polynomial `p`, that is the largest variable with respect to the ranking of `parent(p)`. If `p` is constant, an error is raised.
+"""
+function leader(apre::ActionPolyRingElem)
+  @req !is_constant(apre) "A constant polynomial has no leader"
+  if is_univariate(apre)
+    return vars(apre)[1]
+  end
+  return max(vars(apre)...)
```

Comment:
```suggestion
  return maximum(vars(apre))
```
this is more efficient